golem-component-service-base/tests/db/migration/sqlite/001__init.sql

```diff
@@ -0,0 +1,14 @@
+CREATE TABLE components
+(
+    namespace           text    NOT NULL,
+    component_id        uuid    NOT NULL,
+    name                text    NOT NULL,
+    size                integer NOT NULL,
+    version             bigint  NOT NULL,
+    created_at          timestamp without time zone DEFAULT CURRENT_TIMESTAMP NOT NULL,
+    user_component      text    NOT NULL,
+    protected_component text    NOT NULL,
+    protector_version   bigint,
+    metadata            blob    NOT NULL,
+    PRIMARY KEY (namespace, component_id, version)
```

Comment:
Consider normalising the schema - create 2 tables: `components` with `component_id` primary key and `component_versions` with `(component_id, version)` primary key and a `component_id` foreign key. `namespace` and name should be in `components`.

Current implementation allows different names for different versions. It also requires backend check for unique `namespace`.